tgui/packages/tgui/interfaces/Workshop.js

```diff
@@ -0,0 +1,247 @@
+import { createSearch, toTitleCase } from 'common/string';
+import { useBackend, useLocalState } from "../backend";
+import { Box, Button, Collapsible, Dropdown, Flex, Input, LabeledList, ProgressBar, Section } from '../components';
+import { Countdown } from '../components/Countdown';
+import { Window } from "../layouts";
+
+const canBeMade = (design, brsail, pwrail) => {
+  if (design.requirements === null) {
+    return true;
+  }
+  if (design.requirements["brass"] > brsail) {
+    return false;
+  }
+  if (design.requirements["power"] > pwrail) {
+    return false;
+  }
+  return true;
+};
+
+export const Workshop = (_properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    brass_amount,
+    power_amount,
+    building,
+    buildStart,
+    buildEnd,
+    worldTime,
+  } = data;
+
+  let brassReadable = brass_amount.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,'); // add thousands seperator
```

Comment:
`brassReadable` не меняется, поэтому её нужно объявлять как константу.
```suggestion
  const brassReadable = brass_amount.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,'); // add thousands seperator
```